### ISSUE TITLE
Grading Scheme

### ISSUE THREAD

**Issue by BachDinh1994** [usc-csci201-fall2013/simcity201#71]
@alghanmi @dwilczyn @crowley1958 

SimCity is no doubt a very interesting project by all means. It exercises our creativity, teamwork skills, and individual skills to a great extent. It's certainly not easy, but it's definitely useful. However, my concern is that I do not believe the current grading scheme reflects the final grade that each student should receive. For example, the same percentage given for individual score and team score is not so fair. Because this is the first intensive engineering team project for most cs201 students, it's normal to have communication and team issues. Professor Wilczynski did an excellent presentation giving us advice and tips on how to behave and lead a successful team. Nevertheless, it's difficult to actually have a successful team even with those tips when it's your first time. In most teams, some members will work a lot more than others, so they should deserve more credit, but are still held accountable for their group, just not as much as their individual contribution. Therefore, I believe the grading scheme should have 30-35% individual score (roughly 15-17.5% for each deliverable) and 15-20% team score (roughly 7.5-10%). Or, for the first deliverable, it could be 17.5% individual and 7.5% team because it's our first time. Because we will be more experienced on teamwork for the second deliverable, it can go back to 12.5% individual and 12.5% team. Please excuse the long post. 

PS: This is not only for the benefit of my team members and myself, but also for those who work hard in their teams and deserve credit for what they've done.

**Comment by waynchi** [comment]
I think this is a personal Issue. Some people in our group have finished their portions, some have not. However, I do not believe that it is because certain people in my group are vastly better coders than the others. Rather, I believe it is because of the varying difficulty in tasks. For example, since our group has no one who took CS200, the Gui is definitely one of the hardest parts. On the other hand, the banking system, is not as complex of a task. 

Yet, because the person assigned to our banking system finished earlier, he is now helping out on the Gui. In fact, we're all slowly moving to finish to finish the Gui together. It's a team effort; therefore it should be a team grade. It would not be fair for the person doing the Banking system to get an A while the person who does the Gui to get a C because the bank is complete and the Gui is not.

**Comment by BachDinh1994** [comment]
I believe that's why I suggested for the first deliverable, it should be 17.5% individual and 7.5% team instead of 25% individual and 0% team. That's ONLY for the first deliverable as you can see. I sincerely agree with the grading scheme for the second iteration because we will have been more experienced then. Plus, the graders are smart and capable individuals, so they can obviously tell that those who did the bank had an easier time than those who did the GUI.

**Comment by johngtimms** [comment]
I think that the grading for large team projects will be **inherently** unfair in any situation where all members are not able to commit themselves equally (due to classes, jobs, and extracurriculars) to the project. In essence, it's the night before the due date so spend your time _getting work done_ and take up your issues with your professor directly _after_ you've gotten your grade back.

**Comment by waynchi** [comment]
Yo yea. And it's not 25% individual 0% team... it's 12.5% for each.

**Comment by BachDinh1994** [comment]
It's a suggestion waynchi. I suggested 17.5% individual and 7.5% team as opposed to 25% and 0% team because teamwork still counts to a certain extent. Almost everyone who takes cs201 obviously knows it's 12.5% each.

johngtimms, thank you for your wonderful advice that I'd never thought of ever since I started working on this project.

**Comment by alghanmi** [comment]
tl;dr &mdash; trust us, we have your best interest at heart.

I have been around teams projects for throughout my academic career in different capacities: member, manager, mentor and TA, and I can say - without a doubt - that the best thing you can do in a team project is: **forget about the grades**.

The way team projects work, they evolve from being _an assignment_ to be _the pride of my semester_ . Do not approach your project as a lawyer where their goal is increase the billable hours to get a bigger payday (vis-à-vis grades), but approach it as an engineer who the challenge and sense of accomplishment is the biggest reward. Trust me, if that is your goal, your grade **will** (yes **WILL**) follow.

Think of it this way: when it is all said and done, are you **proud** (not OK, but real unequivocal pride) to put your name on this project after seeing your competition? If your answer is yes, I am pretty sure your grades will be very satisfactory.

Every team has three kinds of members:
1. The one(s) &ndash; those are the people who go above and beyond the call of duty when it comes to the project. It becomes their food, their drink and their sleep.
2. Goldilocks &ndash; these are the members who complete their tasks as specified by the team, no more, no less; just right.
3. Humpty but not Dumpty &ndash; these students usually requires more assistance from their team and may not be as comfortable with code as (1) and (2). The team usually, adapts, helps these members and assign them appropriate tasks. It is always a good idea to pair a (3) with a (2).
4. NoShow/NoCode &ndash; these should be reported to the teaching staff ASAP

At the end of the project, each team will evaluate its members and usually all students fall within one of the four categories above. In 95% of the cases, (1) and (2) get the grades they expect even if the team did not produce a very polished product.

Let me take my TA hat off and put my student hat on and give you advise from a person who has been in academia his whole life (literally, I was born in a university hospital to univ. faculty parents). Grades are important, but they are not the end game. In good institutions such as USC, the people who handle your grades are competent folks who do not take any pride in taking away points. They use scores as a teaching tool just like the whiteboard is a tool. Also, they are fair and admit when they are wrong, they do what is right and give you a way to express any misgivings you may have. The professors in this course make sure of that.

Keep in mind that fairness in this context means treating everyone, i.e. grading everyone, the same way with no favoritism. We be objective as much as possible and only become subjective within the confines of a defined set of guidelines. 

There are a number of behaviors in relation to grades that as a student you must refrain from. They do not do you any favors and most certainly they are bad habits to develop and maintain. To put it frankly, if you want the professor and/or teaching staff member to respect you or you expect to ask them for a reference, you must avoid being:
1. The Nickel and Dimer: a students who runs after the Professor/TA/CP and takes a great deal of time and resources for a 0.5 here and 0.25 there. In the grand scheme of things, these are negligible. Do the math. This is just a waste of time and trust me, you lose all credibility.
2. The Heckler: Whoever is teaching you, most likely studied this subject matter at least before the first smart-phone was made. They have a very good grasp of the concepts and developed an instinct for these things. You should accept their judgement of what is right and what is wrong. You are within your rights to challenge it, but you need a very good argument or even better a scholarly source to back you up.
3. The Gangsta: As a student, you are selected from a huge crop of applicants and you were deemed worthy for this school. You spend here 4 years and leave. Now think of professors, they are selected from as big a pool, but they are hired to stay. Don't you think the university will put in the same due diligence and a very tough and rigorous selection process for them? Then, **show some respect**. Words like ridiculous, absurd and profanities are completely unbecoming and will get you no where so is raising your voice or causing mayhem. If you feel strongly about a matter, discuss it in a dignified and academic fashion in person or in writing.
4. The Aflac Duck: don't be the person who clings to other people's mistakes and parade's when a problem arises.
5. The Complainer: do not be the person who the professor expects to the first in their door whenever a problem arises. Try to deal with things.

To conclude, what you are experiencing in 201 is an academic challenge that should be dealt with in academic way. You may have not been used to this much pressure or this way of specifying a problem, but as a Trojan it is your duty to work at it and get it done. This is **your last programming class**. You will _not_ get any other _real_ software development experience the rest of your academic career at USC. If the first two letters of your major as `CS`, you need to make the best of this experience.

**Comment by BachDinh1994** [comment]
Thank you for your comment Rami
